Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLMultiProcess.py

```diff
@@ -0,0 +1,990 @@
+                             doc='Empty beam run(s).')
+        self.setPropertyGroup('EmptyBeamRuns', 'Numors')
+
+        self.declareProperty(MultipleFileProperty(name='FluxRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Empty beam run(s) for flux calculation only; '
+                                 'if left blank the flux will be calculated from EmptyBeamRuns.')
+        self.setPropertyGroup('FluxRuns', 'Numors')
+
+        self.declareProperty(MultipleFileProperty(name='EmptyContainerRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Empty container run(s).')
+        self.setPropertyGroup('EmptyContainerRuns', 'Numors')
+
+        #================================TR INPUT RUNS================================#
+
+        for l in range(N_LAMBDAS):
+            p_name = f'SampleTrRunsW{l+1}'
+            self.declareProperty(MultipleFileProperty(name=p_name,
+                                                      action=FileAction.OptionalLoad,
+                                                      extensions=['nxs'],
+                                                      allow_empty=True),
+                                 doc=f'Sample transmission run(s) at the wavelength #{l+1}.')
+            self.setPropertyGroup(p_name, 'Transmission Numors')
+
+        self.declareProperty(MultipleFileProperty(name='TrDarkCurrentRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Absorber (Cd/B4C) run(s) for transmission calculation.')
+
+        self.declareProperty(MultipleFileProperty(name='ContainerTrRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Container transmission run(s).')
+        self.setPropertyGroup('ContainerTrRuns', 'Transmission Numors')
+
+        self.declareProperty(MultipleFileProperty(name='TrEmptyBeamRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Empty beam run(s) for transmission calculation.')
+
+        self.setPropertyGroup('TrDarkCurrentRuns', 'Transmission Numors')
+        self.setPropertyGroup('ContainerTrRuns', 'Transmission Numors')
+        self.setPropertyGroup('TrEmptyBeamRuns', 'Transmission Numors')
+
+        #=============================INPUT FILES/WORKSPACES==============================#
+
+        self.declareProperty(name='SensitivityMap', defaultValue='',
```

Comment:
If I enter a nonsense string for any of these "input files\workspaces" properties in the UI and run the algorithm, the dialog closes but an error is reported from the LoadNexusProcessed algorithm. Could these be validated in the dialog?

Reply:
Ideally there should be File/Workspace property to handle all of that. But in the absence thereof, maybe I could use the `FileFinder` to validate that.